### PR TITLE
Tweak: implemented virtual list render in Icons Library

### PR DESCRIPTION
**Pull request by bainternet**
boost Icons library performance in 6000%

### CODE REVIEW

**Review by danielkatz — APPROVED** [review]
Looking great.
Maybe better let React worry about subscribing and unsubscribing to the scroll event?

**Comment by bainternet** [comment]
@danielkatz 

meaning?

**Comment by danielkatz** [comment]
Is it possible to use `onScroll={someScrollHandler}` instead on manually adding and removing 'scroll' event listener with (add|remove)EventListener?

**Comment by bainternet** [comment]
Oh that, well it's out of context and on a higher level wrapper parent

File: assets/dev/js/editor/components/icons-manager/components/icon-list.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+	attachScrollListener = () => {
+		const element = document.getElementById( this.selectors.wrapper );
+		if ( element ) {
+			element.addEventListener( 'scroll', this.handleScroll );
+		}
+	};
+
+	maybeMeasureItem = () => {
+		if ( ! this.state.itemSize.width ) {
```
Comment (KingYes) [inline assets/dev/js/editor/components/icons-manager/components/icon-list.js]: reverse logic

File: assets/dev/js/editor/components/icons-manager/components/icon-list.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+			const wrapper = document.getElementById( this.selectors.wrapper );
+			const testElement = document.querySelector( this.selectors.item );
+			if ( ! testElement ) {
+				return;
+			}
+			const newState = {
+				itemSize: {
+					width: testElement.offsetWidth + 20,
```
Comment (KingYes) [inline assets/dev/js/editor/components/icons-manager/components/icon-list.js]: use variable for padding size

File: assets/dev/js/editor/components/icons-manager/components/icon-manager.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ import Tab from './tab';
 import IconsGoPro from './icons-go-pro';
 
 class IconsManager extends Component {
+	scrollViewRef = React.createRef();
```
Comment (KingYes) [inline assets/dev/js/editor/components/icons-manager/components/icon-manager.js]: import react

**Review by KingYes — CHANGES_REQUESTED** [review]

**Review by KingYes — APPROVED** [review]

**Pull request merged**